src/common/components/BaniList/baniOrderHelper.js

```diff
@@ -7,25 +9,44 @@ const extractBaniDetails = (baniItem) => {
   };
 };
 const orderedBani = (baniList, baniOrder) => {
-  const defaultBaniOrder = baniOrder;
-  const banis = [];
-  if (defaultBaniOrder && defaultBaniOrder.baniOrder.length > 0) {
-    defaultBaniOrder.baniOrder.forEach((element) => {
-      if (element.id) {
-        const baniItem = findBaniById(baniList, element.id);
-        if (baniItem) {
-          banis.push(extractBaniDetails(baniItem));
-        }
-      } else {
-        const folder = element.folder.map((item) => {
-          const bani = findBaniById(baniList, item.id);
-          return extractBaniDetails(bani);
-        });
-        banis.push({ gurmukhi: element.gurmukhi, translit: element.translit, folder });
-      }
-    });
+  const isValidBaniOrder =
+    baniOrder != null && // checks for null or undefined
+    typeof baniOrder === "object" &&
+    Array.isArray(baniOrder.baniOrder) &&
+    baniOrder.baniOrder.length > 0;
```

Comment:
I came across same code previously as well in this PR. Can we define a function to check this, if its' going to be used at multiple places?